src/aws_encryption_sdk/internal/crypto/elliptic_curve.py

```diff
@@ -183,8 +183,7 @@ def generate_ecc_signing_key(algorithm):
     :returns: Generated signing key
     :raises NotSupportedError: if signing algorithm is not supported on this platform
     """
-    try:
-        verify_interface(ec.EllipticCurve, algorithm.signing_algorithm_info)
-        return ec.generate_private_key(curve=algorithm.signing_algorithm_info(), backend=default_backend())
-    except InterfaceNotImplemented:
-        raise NotSupportedError("Unsupported signing algorithm info")
+    for method in ec.EllipticCurve.__abstractmethods__:
+        if not hasattr(algorithm.signing_algorithm_info(), method):
+            raise NotSupportedError("Unsupported signing algorithm info")
```

Comment:
Issue/Blocking: You MUST NOT copy and paste code.
Abstract and simplify: author `verify_interface` in 
`src/aws_encryption_sdk/internal/utils/__init__.py`.

Reference it here and in `src/aws_encryption_sdk/internal/crypto/authentication.py`.